### PR TITLE
Added more UI improvements

### LINKED ISSUE #2309: Automation failure : test_sync_rh_repos
TestNames:
tests.foreman.ui.test_sync.Sync.test_sync_rh_repos

Error Message
Could not select the product: 'Red Hat Enterprise Linux Server'

Stacktrace

  File "/usr/lib64/python2.7/unittest/case.py", line 369, in run
    testMethod()
  File "/home/jenkins/workspace/satellite6-automation-downstream-rhel66-ui/tests/foreman/ui/test_sync.py", line 83, in test_sync_rh_repos
    self.sync.enable_rh_repos(repos)
  File "robottelo/ui/sync.py", line 162, in enable_rh_repos
    "Could not select the product: '{0}'".format(PRDS[prd]))
'Could not select the product: \'Red Hat Enterprise Linux Server\'\n--------------------

### PR DESCRIPTION
**Pull request by oshtaier**
Fixes refer to `ui.hardwaremodel`, `ui.location`, `ui.products`, `ui.sync` and `ui.user` modules.
 Most work done around #2562 issue and our style in general

 Closes #2309. That defect was reproducing on Jenkins for last 6 months. (at least)

### CODE REVIEW

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/ui/test_hw_model.py 
...............................

----------------------------------------------------------------------
Ran 30 tests in 1235.476s

OK
```
```
nosetests tests/foreman/ui/test_location.py
.....................................................................................................................................................

----------------------------------------------------------------------
Ran 147 tests in 6115.320s

OK
```
```
nosetests tests/foreman/ui/test_sync.py
........

----------------------------------------------------------------------
Ran 8 tests in 573.917s

OK
```
```
nosetests tests/foreman/ui/test_products.py
....................................................

----------------------------------------------------------------------
Ran 52 tests in 2067.337s

OK
```
```
nosetests tests/foreman/ui/test_user.py
..........SSSSSSSSSSSSSSSS.......................SSSSSSSSSSSSSSSSSSSSSSSSSSSSS.S............................................S.....SSSSSSSSSSSSSSSSSSSSSSSSSSSS.........SSSSS..

----------------------------------------------------------------------
Ran 173 tests in 2396.732s

OK (SKIP=80)
```

**Comment by sghai** [comment]
Awesome. Nice to see test results. thank you

**Comment by oshtaier** [comment]
For double confirmation, re-run on 7.1:
```
nosetests tests/foreman/ui/test_sync.py
........
----------------------------------------------------------------------
Ran 8 tests in 832.303s

OK
```

**Comment by elyezer** [comment]
ACK

File: robottelo/ui/base.py [hunk 1]
```diff
@@ -458,7 +459,7 @@ def click(self, locator, wait_for_ajax=True, timeout=30):
         :raise: UINoSuchElementError if the element could not be found.
```
Comment (rplevka) [inline robottelo/ui/base.py]: Could you update the docstring accordingly?

File: tests/foreman/ui/test_location.py [hunk 2]
```diff
@@ -276,53 +242,53 @@ def test_add_domain_1(self, domain_name):
...
             self.assertIsNotNone(self.location.search(loc_name))
             self.location.update(loc_name, new_users=[user_name])
             self.location.search(loc_name).click()
-            session.nav.wait_until_element(
-                tab_locators["context.tab_users"]).click()
-            element = session.nav.wait_until_element((strategy,
-                                                      value % user_name))
+            session.nav.click(tab_locators['context.tab_users'])
+            element = session.nav.wait_until_element(
+                (strategy, value % user_name))
```
Comment (rplevka) [inline tests/foreman/ui/test_location.py]: nitpick: I __guess__ this should be the correct formatting

```python
element = session.nav.wait_until_element(
    (strategy, value % user_name)
)
```
...ignore, if you're sure I'm not right

File: robottelo/ui/base.py (hunk 1 above)
Comment (oshtaier) [inline robottelo/ui/base.py]: :+1: 
Fixed

**Comment by rplevka** [comment]
Overall, the changes seem sane to me.
I like the way `.click()` simplifies the code

ACK pending minor comments

File: tests/foreman/ui/test_location.py (hunk 2 above)
Comment (oshtaier) [inline tests/foreman/ui/test_location.py]: That's correct one according to pep8

Comment (rplevka) [inline tests/foreman/ui/test_location.py]: :+1:

**Pull request merged**